src/blockmap.jl

```diff
@@ -297,75 +297,96 @@ LinearAlgebra.transpose(A::BlockMap) = TransposeMap(A)
 LinearAlgebra.adjoint(A::BlockMap)  = AdjointMap(A)
 
 ############
-# multiplication with vectors
+# multiplication helper functions
 ############
 
-function A_mul_B!(y::AbstractVector, A::BlockMap, x::AbstractVector)
-    require_one_based_indexing(y, x)
-    m, n = size(A)
-    @boundscheck (m == length(y) && n == length(x)) || throw(DimensionMismatch("A_mul_B!"))
+@inline function _blockmul!(y, A::BlockMap, x, α, β)
     maps, rows, yinds, xinds = A.maps, A.rows, A.rowranges, A.colranges
     mapind = 0
     @views @inbounds for rowind in 1:length(rows)
-        yrow = y[yinds[rowind]]
+        yrow = selectdim(y, 1, yinds[rowind])
         mapind += 1
-        A_mul_B!(yrow, maps[mapind], x[xinds[mapind]])
+        mul!(yrow, maps[mapind], selectdim(x, 1, xinds[mapind]), α, β)
         for colind in 2:rows[rowind]
             mapind +=1
-            mul!(yrow, maps[mapind], x[xinds[mapind]], true, true)
+            mul!(yrow, maps[mapind], selectdim(x, 1, xinds[mapind]), α, true)
         end
     end
     return y
 end
 
-function At_mul_B!(y::AbstractVector, A::BlockMap, x::AbstractVector)
-    require_one_based_indexing(y, x)
-    m, n = size(A)
-    @boundscheck (n == length(y) && m == length(x)) || throw(DimensionMismatch("At_mul_B!"))
+@inline function _transblockmul!(y, A::BlockMap, x, α, β, transform)
```

Comment:
This is corresponding multiplication code we used to have twice, once for `At_mul_B!` and once for `Ac_mul_B!`. Otherwise, this is the analogous generalization to alpha, beta, and matrices.